tools/test-locked-requirements

```diff
@@ -30,30 +34,80 @@ def print_diff(path_file1, path_file2):
 
-    if not same:
-        print_diff(TEST_PROD_LOCK_FILE, PROD_LOCK_FILE)
-        print_diff(TEST_DEV_LOCK_FILE, DEV_LOCK_FILE)
-        # Flush the output to ensure we print the error at the end.
-        sys.stdout.flush()
-        raise Exception("Seems like you have updated some python dependencies but haven't "
-                        "run updated requirements lock files. Please update them by running "
-                        "`tools/update-locked-requirements`. For more information please refer "
-                        "to `requirements/README.md`.")
+    return same
+
+def get_requirements_hash(use_test_lock_files=False):
+    # type: (Optional[bool]) -> str
+    sha1 = hashlib.sha1()
+    req_files = glob.glob(os.path.join(ZULIP_PATH, "requirements", "*.in"))
+    lock_files = (DEV_LOCK_FILE, PROD_LOCK_FILE)
+    if use_test_lock_files:
+        lock_files = (TEST_DEV_LOCK_FILE, TEST_PROD_LOCK_FILE)
+    req_files.extend(lock_files)
+    for file_path in req_files:
+        with open(file_path) as fp:
+            sha1.update(fp.read().encode("utf-8"))
+    return sha1.hexdigest()
+
+def may_be_setup_cache():
+    # type: () -> None
+    if not os.path.exists(CACHE_DIR):
+        subprocess.check_call(['mkdir', CACHE_DIR])
+    if not os.path.exists(CACHE_FILE):
+        with open(CACHE_FILE, 'w') as fp:
+            ujson.dump([], fp)
+
+def load_cache():
+    # type: () -> List[str]
+    with open(CACHE_FILE) as fp:
+        hash_list = ujson.load(fp)
+    return hash_list
+
+def update_cache(hash_list):
+    # type: (List[str]) -> None
+    # We store last 50 hash entries. Aggressive caching is
+    # not a problem as it is cheap to do.
+    if len(hash_list) > 50:
+        hash_list = hash_list[-50:]
+    with open(CACHE_FILE, 'w') as fp:
+        ujson.dump(hash_list, fp)
+
+def main():
+    # type: () -> None
+    may_be_setup_cache()
+    hash_list = load_cache()
+    curr_hash = get_requirements_hash()
+    if curr_hash not in hash_list:
+        requirements_are_consistent = test_locked_requirements()
+        # Cache the hash so that we need not to run the `update_locked_requirements`
+        # tool again for checking this set of requirements.
+        valid_hash = get_requirements_hash(use_test_lock_files=True)
+        hash_list.append(valid_hash)
+        update_cache(hash_list)
+        if not requirements_are_consistent:
+            print_diff(TEST_PROD_LOCK_FILE, PROD_LOCK_FILE)
+            print_diff(TEST_DEV_LOCK_FILE, DEV_LOCK_FILE)
+            # Flush the output to ensure we print the error at the end.
+            sys.stdout.flush()
+            raise Exception("Seems like you have updated some python dependencies but haven't "
+                            "updated locked requirements files. Please update them by running "
+                            "`tools/update-locked-requirements`. For more information please "
+                            "refer to `requirements/README.md`.")
```

Comment:
I think you want to invert some of these if statements and return, rather than having things get so indented.

Reply:
Also, "It looks like you" is better English than "Seems like you ...".